### PR TITLE
Migrate cray-product-catalog ConfigMap to multiple product ConfigMaps

### PR DESCRIPTION
**Pull request by lathanm**
## Summary and Scope

Migrate cray-product-catalog ConfigMap to multiple product ConfigMaps before upgrading cray-product-catalog to 1.10.0 which puts the cray-product-catalog data into main and associated product ConfigMap

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASM-4546](https://jira-pro.it.hpe.com:8443/browse/CASM-4546), [CASM-4545](https://jira-pro.it.hpe.com:8443/browse/CASM-4545)
* Change will also be needed in ``
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing

Tested on local system and mug system as an independent python script

### Tested on:

  * `<mug-ncn-m001.hpc.amslabs.hpecorp.net>`
  * Local development environment

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why? yes
- Was downgrade tested? If not, why? no
- Were new tests (or test issues/Jiras) created for this change? yes

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ X] Target branch correct
- [X ] Testing is appropriate and complete, if applicable
- [X ] Is a new version being released? Update https://github.com/Cray-HPE/cray-product-catalog/wiki/CSM-Compatibility-Matrix
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by arnab-muk-hpe — APPROVED** [review]
We will write unit test cases post that.
@lathanm

File: cray_product_catalog/migration/config_map_data_handler.py [hunk 1]
```diff
@@ -166,5 +129,53 @@ def migrate_config_map_data(self, config_map_data):
...
+                else:
+                    LOGGER.error("Failed to delete ConfigMap %s, retrying..", rename_from)
+                    del_failed = True
+                    break
+            else:
+                LOGGER.error("Failed to create ConfigMap %s, retrying..", rename_to)
+                continue
+        # Since only delete of backed up ConfigMap failed, retrying only delete operation
+        if del_failed:
+            while attempt < 10:
```
Comment (krabhi-hpe) [inline cray_product_catalog/migration/config_map_data_handler.py:171]: `attempt` should be reset to `0` before this ?

**Review by krabhi-hpe — CHANGES_REQUESTED** [review]

File: tests/migration/test_config_map_data_handler.py [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+    PRODUCT_CATALOG_CONFIG_MAP_NAME, PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE,
+    PRODUCT_CATALOG_CONFIG_MAP_LABEL
+)
+from cray_product_catalog.util.catalog_data_helper import format_product_cm_name
+from tests.mock_update_catalog import (
+    ApiInstance, ApiException
+)
+from cray_product_catalog.migration import CONFIG_MAP_TEMP
+
+YAML_DATA = """
```
Comment (arnab-muk-hpe) [inline tests/migration/test_config_map_data_handler.py:50]: shall we move these to separate file `migration_mock.py`? or we will keep it here?
@krabhi-hpe @lathanm

File: tests/migration/test_config_map_data_handler.py [hunk 3]
```diff
@@ -0,0 +1,744 @@
...
+            self.assertEqual(
+                captured.records[0].getMessage(),
+                f"Created product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[0]}")
+
+            self.assertEqual(
+                captured.records[1].getMessage(),
+                f"Created product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[1]}")
+
+    def test_create_second_product_config_map_failed(self):
+        """ Validating failed to create product config maps """
```
Comment (arnab-muk-hpe) [inline tests/migration/test_config_map_data_handler.py:250]: change the doc string to address second configmap

File: tests/migration/test_config_map_data_handler.py [hunk 4]
```diff
@@ -0,0 +1,744 @@
...
+            self.assertEqual(
+                captured.records[0].getMessage(),
+                f"Created product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[0]}")
+
+            self.assertEqual(
+                captured.records[1].getMessage(),
+                f"Failed to create product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[1]}")
+
+    def test_create_first_product_config_map_failed(self):
+        """ Validating failed to create product config maps """
```
Comment (arnab-muk-hpe) [inline tests/migration/test_config_map_data_handler.py:284]: change the doc string to address first configmap

**Review by arnab-muk-hpe — CHANGES_REQUESTED** [review]
Few cases we need to add
- empty main config map -> exit
- main config map read failed -> exist 
- Already splited cms i.e config version not present in main -> exit 
shall complete all cases.

Others please look @lathanm @krabhi-hpe @paulose-kuriakose @anoop1402

File: tests/migration/test_config_map_data_handler.py (hunk 4 above)
Comment (lathanm) [reply tests/migration/test_config_map_data_handler.py]: changed

File: tests/migration/test_config_map_data_handler.py (hunk 2 above)
Comment (lathanm) [reply tests/migration/test_config_map_data_handler.py]: moved

**Comment by lathanm** [comment]
> * ig map read failed -> exist
> * Already splited cms i.e confi

Added all these test cases

**Review by arnab-muk-hpe — APPROVED** [review]

**Review by paulose-kuriakose — APPROVED** [review]

**Review by atifsyedali — APPROVED** [review]

**Pull request merged**